src/Metadata/SnapshotMetadata.php

```diff
@@ -29,9 +28,13 @@ protected static function getSignedCollectionOptions(): array
             new Count(['min' => 1]),
             new All([
                 new Collection(
-                    static::getVersionConstraints() +
                     [
-                        'length' => new Optional([new Type('integer')]),
+                        'fields' => static::getVersionConstraints(),
+                        // These fields are mentioned in the specification as optional but the Python library does not
+                        // adding these fields. Since we use the Python library for our fixtures we cannot create test
+                        // fixtures that have these fields specified.
+                        'unsupportedFields' => ['length', 'hashes'],
```

Comment:
So...the Python library doesn't add them, but this comment implies that they *are* part of the spec. Do we really want to mark them as unsupported? Maybe we should at least have a TODO here?

Reply:
They are optional in the spec 

> The snapshot.json file is signed by the snapshot role. It MUST list the version numbers of the top-level targets metadata and all delegated targets metadata. It MAY also list their lengths and file hashes.

It think if the Python library doesn't add optional fields we shouldn't support them because we don't have way to make test fixtures for them. But see the ongoing discussion here #97